parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java

```diff
@@ -578,14 +608,8 @@ public ColumnChunkPageReader readAllPages() throws IOException {
             if (dictionaryPage != null) {
               throw new ParquetDecodingException("more than one dictionary page in column " + descriptor.col);
             }
-          DictionaryPageHeader dicHeader = pageHeader.getDictionary_page_header();
-          dictionaryPage =
-                new DictionaryPage(
-                    this.readAsBytesInput(compressedPageSize),
-                    uncompressedPageSize,
-                    dicHeader.getNum_values(),
-                    converter.getEncoding(dicHeader.getEncoding())
-                    );
+
+            dictionaryPage = readDictionary(configuration, pageHeader, descriptor.metadata, f);
```

Comment:
This no longer calls readAsBytesInput, which gets the page data out of an existing buffer. It also decompresses the dictionary page, which I don't think was happening before. I've been editing this a bit, and I think things end up cleaner if we don't reuse the readDictionary method here.